CSV Export Changes

leorudczenko opened the pull request:
## Description

This pull request updates the formatting which is done upon CSV export:
- The `id` and `sample_name` columns are now concatenated into a single column `Name`.
	- A separator `_#` is used in the new `Name` column to identify the start/end of the concatenated values.
- `id` values are now padded using leading `0` characters to ensure the `id` value contains 3 digits.

**Backwards compatibility is supported for CSV files which do not include these new formatting features.**

Following a demonstration meeting with Matt, a couple of small changes have been made to the interface when in scaling mode:
- The `SetScaleDialog` box now opens at the top left of the main window.
- All main window input widgets other than the `GraphicsScene` are now disabled when TACtool is in scaling mode.

_The expected test data has been updated to reflect these changes and test their functionality._

leorudczenko committed:
Update CSV export to concatenate 'sample_name' and 'id' into the column 'Name'

leorudczenko committed:
Add zero padding for analysis point id values upon CSV export

leorudczenko committed:
Improve usability when using scaling mode

volcan01010 committed:
Make scale_value_input read-only

Having the "set scale" beside the line-edit for scale value is confusing
because it implies that you press that button to confirm the changes.
Users will not know the scale to enter it directly, so it is easiest to
remove the option.

volcan01010 commented:
I tested this locally and all worked.  I made a minor tweak so that the set_scale_line_edit is read-only.

leorudczenko committed:
Update scaling instructions to reflect changes

leorudczenko committed:
Add typehint

leorudczenko committed:
Update method used to disable scale_value_input

leorudczenko committed:
Minor scaling window changes for usability

leorudczenko committed:
Update class relationship diagram to reflect changes

leorudczenko commented:
I have made a couple more minor changes to clean things up:
- Changed method used to disable `scale_value_input` for consistency
- Removed `scale_value_input` from the list of input fields to change state when setting the scale
- Added the scale units measurement to the `SetScaleDialog` box
- Updated documentation files to reflect the changes
